python/pyarrow/tests/test_cython.py

```diff
@@ -163,6 +163,40 @@ def test_cython_api(tmpdir):
                               env=subprocess_env)
 
 
+@pytest.mark.cython
+def test_extension_type(tmpdir):
```

Comment:
This isn't strictly a cython test, but the cython machinery is useful for generating a C++ Extension Type on the fly. Perhaps this test should be moved elsewhere?

Reply:
I think it is fine either way. It could also be moved to `test_extension_type.py`, and then you can import some of the needed utilities from test_cython.py.

Reply:
Moved to `test_extension_type.py`